File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/sign/SignatureMvpView.java
package com.mifos.mifosxdroid.online.sign;

import com.mifos.api.GenericResponse;
import com.mifos.mifosxdroid.base.MvpView;

/**
 * Created by Tarun on 29-06-2017.
 */

public interface SignatureMvpView extends MvpView {

    void showSignatureUploadedSuccessfully(GenericResponse response);

    void saveAndUploadSignature();

    void requestPermission();

    void checkPermissionAndRequest();

    void showError(int errorId);

    void getDocumentFromGallery();
}

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/sign/SignatureMvpView.kt
package com.mifos.mifosxdroid.online.sign

import com.mifos.api.GenericResponse
import com.mifos.mifosxdroid.base.MvpView

/**
 * Created by Tarun on 29-06-2017.
 */
interface SignatureMvpView : MvpView {
    fun showSignatureUploadedSuccessfully(response: GenericResponse?)
    fun saveAndUploadSignature()
    fun requestPermission()
    fun checkPermissionAndRequest()
    fun showError(errorId: Int)
    val documentFromGallery: Unit
}